lib/AppInfo/Application.php

```diff
@@ -71,6 +73,10 @@ private function initializeAppContainer() {
 
 		$container->registerAlias('ErrorMiddleware', ErrorMiddleware::class);
 		$container->registerMiddleWare('ErrorMiddleware');
+
+		$container->registerService('OCA\Mail\Service\GroupsIntegration', function($c) {
```

Comment:
I don't think you have to register this here. The DI container should be able to resolve this class automatically.

Reply:
How should the container know which `AbstractGroupService`s `GroupsIntegration` should be constructed with?

Reply:
See below. Use an interface instead of this abstract class and register the implementation as alias of the interface here (see other services in this file).

Reply:
Please use `$container->registerAlias(IGroupService::class, NextcloudGroupService::class);` instead